src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetVersionHandler.cs

```diff
@@ -52,5 +75,77 @@ private enum PowerShellProcessArchitecture
             X86,
             X64
         }
+
+        private async Task CheckPackageManagement()
+        {
+            PSCommand getModule = new PSCommand().AddCommand("Get-Module").AddParameter("ListAvailable").AddParameter("Name", "PackageManagement");
+            foreach (PSModuleInfo module in await _powerShellContextService.ExecuteCommandAsync<PSModuleInfo>(getModule))
+            {
+                // The user has a good enough version of PackageManagement
+                if(module.Version >= s_desiredPackageManagementVersion)
+                {
+                    break;
+                }
+
+                _logger.LogDebug("Old version of PackageManagement detected. Attempting to update.");
+
+                var takeActionText = "Yes";
+                MessageActionItem messageAction = await _languageServer.Window.ShowMessage(new ShowMessageRequestParams
+                {
+                    Message = "You have an older version of PackageManagement known to cause issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?",
+                    Type = MessageType.Warning,
+                    Actions = new []
+                    {
+                        new MessageActionItem
+                        {
+                            Title = takeActionText
+                        },
+                        new MessageActionItem
+                        {
+                            Title = "Not now"
+                        }
```

Comment:
I'm not adding a "don't tell me ever again" because:

1. I don't know how to set a setting in the client from the server without adding yet another custom message
2. I kinda wanna make it hard to find the setting to disable this because they really should update their PackageManagement...